src/shared/actions/migrations.js

```diff
@@ -0,0 +1,126 @@
+
+    // Progressbar => (Preparing data)
+    dispatch(setNextStepAsActive());
+
+    return oldStorageAdapter
+        .get()
+        .then((storedData) => {
+            if (isEmpty(storedData)) {
+                throw new Error(Errors.NO_STORED_DATA_FOUND);
+            }
+
+            // Progressbar => Migrating settings
+            dispatch(setNextStepAsActive());
+
+            const settings = get(storedData, 'settings');
+            // If user has wallet settings, migrate them.
+            if (settings) {
+                return migrateSettings(
+                    assign({}, settings, {
+                        // onboardingComplete is part of accounts reducer (in old storage) and has now moved to Wallet schema model
+                        // Assign it to the settings object so that it can be migrated with the rest of the settings.
+                        onboardingComplete: get(storedData, 'accounts.onboardingComplete') || false,
+                        // notificationLog is part of alerts reducer (in old storage) and has now moved to Wallet schema model
+                        // Has also been renamed to errorLog. Assign it to the settings object
+                        errorLog: get(storedData, 'alerts.notificationLog') || [],
+                    }),
+                ).then(() => storedData);
+            }
+
+            return Promise.resolve(storedData);
+        })
+        .then((storedData) => {
+            // Progressbar => Migrating accounts
+            dispatch(setNextStepAsActive());
+
+            const accounts = get(storedData, 'accounts');
+            if (accounts) {
+                return migrateAccounts(accounts).then(() => storedData);
+            }
+
+            return Promise.resolve(storedData);
+        })
+        .then(() => {
+            // Progressbar => Cleaning old data
+            dispatch(setNextStepAsActive());
+
+            return oldStorageAdapter.clear();
+        })
+        .then(() => {
+            // Progressbar => Migration Complete
+            dispatch(setNextStepAsActive());
+
+            setTimeout(onComplete, 2000);
+        })
+        .catch((error) => {
+            if (error.message === Errors.NO_STORED_DATA_FOUND) {
+                // If there is not data to migrate, just mark migration as complete
+                onComplete();
+            } else {
+                dispatch(
+                    generateAlert(
+                        'error',
+                        i18next.t('global:somethingWentWrong'),
+                        i18next.t('migration:problemMigratingYourData'),
+                        5500,
+                        error,
+                    ),
+                );
+            }
+        });
+};
+
+/**
+ * Dispatch to set migration (AsyncStorage to Realm) complete
+ *
+ * @method setMigrationStatus
+ * @param {bool} payload
+ *  *
+ * @returns {{type: {string}, payload: {bool} }}
+ */
+export const setMigrationStatus = (payload) => {
```

Comment:
Maybe best to use `setRealmMigrationStatus` here and update the associated variable names. We use the term `migration` for a couple of things.

Reply:
Fixed in https://github.com/iotaledger/trinity-wallet/pull/375/commits/4198d97852e19380c8d557b68a58a6a1a8490375